plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -61,11 +66,50 @@ protected void setup(Binder binder)
                 .setDefault()
                 .toInstance(SIMPLE_USER_NAME_PROVIDER);
 
+        install(conditionalModule(
+                ThriftHttpMetastoreConfig.class,
+                config -> config.getAuthenticationMode()
+                        .filter(isEqual(BEARER))
+                        .isPresent(),
+                conditionalBinder -> configBinder(conditionalBinder).bindConfig(ThriftHttpMetastoreConfig.class)));
+
         binder.bind(Key.get(boolean.class, AllowHiveTableRename.class)).toInstance(true);
 
+        validateThriftHttpConfiguration();
         install(new ThriftMetastoreAuthenticationModule());
     }
 
+    private void validateThriftHttpConfiguration()
```

Comment:
This logic needs improvement. The implementation is very fragile for future changes. 
When we add a new config property to ThriftMetastoreConfig, we must visit this method (I believe we will forget) and confirm if additional validation is required or not.